### PR TITLE
Integrated micrometer metering #774

### PR DESCRIPTION
**Pull request by pravussum**
This is the initial draft of the micrometer integration to address #774.

I figured the best place to put it would be the already existing core.io.monitoring bundle. I'll move it otherwise.

Some notes: 
- the bundle registers some "default" metrics (basically all JVM related) and some openhab specific metrics (ThreadManager pool statistics, event counts, bundle states, thing states, rule run counts)
- it uses the Micrometer [global meter registry](https://micrometer.io/docs/concepts#_global_registry), a [composite registry](https://micrometer.io/docs/concepts#_composite_registries) where other registries (e. g. the Prometheus one) can attach later to generate the actual metrics
- I'm developing an IO addon, which attaches to the global meter registry and provides the actual metrics as REST endpoint (PR https://github.com/openhab/openhab-addons/pull/9890)
- other new Metrics could easily be added by registering with the global meter registry at their own convenience within their respective bundle

Some questions:
- I had a hard time to get the Maven dependencies, bundle dependencies and Karaf features straight. Is this the right way to include a third party Maven dependency (non OSGI-ready) including its transitive dependencies? See https://community.openhab.org/t/3rd-party-maven-dependencies-in-openhab-core/113722 for a related forum post as well. I ended up using Karafs wrap feature instead of adding the dependency to the osgiify repo - although no one else seems to have used it that way. 
 
/cc @kaikreuzer

### CODE REVIEW

**Comment by wborn** [comment]
I saw a Micrometer presentation once and ever since put it on my todo list to start integrating it into projects. So thanks for creating this PR. :+1:

File: features/karaf/openhab-tp/src/main/feature/feature.xml
Comment (wborn) [inline features/karaf/openhab-tp/src/main/feature/feature.xml:184]: Using the wrap protocol causes issues when for instance debugging the code in non-Karaf environments. E.g. it won't work when running the Demo App in Eclipse. To fix that you can either use bnd or Maven to embed a non-OSGi dependency into your bundle. See for instance the `org.openhab.core.audio` bundle for embedding a dependency in your bundle using Maven.

**Comment by pravussum** [comment]
I think this is in a state where it can be reviewed. Not much action to be seen without the addon PR, though.

**Review by kaikreuzer — APPROVED** [review]
Many thanks @pravussum, this looks awesome to me!
From my perspective, it's ready to merge.
Since I have never worked with micrometer so far, I'd be happy for a second review by @wborn or @cweitkamp, though.

**Comment by kaikreuzer** [comment]
@openhab/core-maintainers If you're not too familiar with it yourself and don't find the time for a review here, it is absolutely fine and I'd just go ahead and merge.

**Comment by cweitkamp** [comment]
I am afraid I will have now time to test it now. Fell free to merge this PR. I will test it afterwards in my prod environment.

**Comment by kaikreuzer** [comment]
Fair enough, let's merge then - thanks @pravussum, looking forward to your work on the add-on side!

**Pull request merged**
